src/platform/common/utils/localize.ts

```diff
@@ -1234,10 +1398,17 @@ export function getCollectionJSON(): string {
 }
 
 // eslint-disable-next-line
-export function localize(key: string, defValue?: string) {
+export function localize(key: string | nls.LocalizeInfo, defaultValue: string) {
     // Return a pointer to function so that we refetch it on each call.
     return () => {
-        return getString(key, defValue);
+        const standardValue = nlsLocalize(key as nls.LocalizeInfo, defaultValue);
+        if (standardValue !== defaultValue) {
+            return standardValue;
+        }
+        if (typeof key !== 'string') {
+            key = key.key;
+        }
+        return getString(key as string, defaultValue);
```

Comment:
This does add a small performance impact, however it should be just constant, since the `vscode-nls` localize function is constant.